src/base/bittorrent/tracker.cpp

```diff
@@ -271,52 +275,95 @@ void Tracker::processAnnounceRequest()
     // ip address
     announceReq.socketAddress = m_env.clientAddress;
     announceReq.claimedAddress = queryParams.value(ANNOUNCE_REQUEST_IP);
+    // request port
+    announceReq.requestPort =  m_env.clientPort;
 
     // Enforce using IPv4 if address is indeed IPv4 or if it is an IPv4-mapped IPv6 address
     bool ok = false;
     const qint32 decimalIPv4 = announceReq.socketAddress.toIPv4Address(&ok);
-    if (ok)
+    if (ok) {
         announceReq.socketAddress = QHostAddress(decimalIPv4);
+        LogMsg(tr(R"(Embedded Tracker: Announce request from non- plain-IPv6 address %1.)"
+                R"( Forcing use of equivalent plain IPv4 address %2.)")
+                .arg(m_env.clientAddress.toString(), announceReq.socketAddress.toString())
+            , Log::INFO);
+    }
 
     // 1. info_hash
     const auto infoHashIter = queryParams.find(ANNOUNCE_REQUEST_INFO_HASH);
-    if (infoHashIter == queryParams.end())
-        throw TrackerError("Missing \"info_hash\" parameter");
```

Comment:
perhaps we should allow exception messages to be translated.

Reply:
Probably a good idea, will handle that after the raw string literals question is resolved.

Reply:
I retract this, error message for peers should stay in English.